src/com/SocketCommunication.cpp

```diff
@@ -199,21 +183,14 @@ void SocketCommunication::requestConnection(std::string const &acceptorName,
   TRACE(acceptorName, requesterName);
   assertion(not isConnected());
 
-  std::string address;
-  const std::string addressFileName("." + requesterName + "-" + acceptorName + ".address");
+  ConnectionInfoReader conInfo(acceptorName, requesterName, _addressDirectory);
+  std::string address = conInfo.read();
+  DEBUG("Request connection to " << address);
+  std::string ipAddress  = address.substr(0, address.find(":"));
```

Comment:
Please use the (`noexcept`) `char` version of `std::string::find` and refactor out the index to prevent the following line from looking so Frankensteiny.
Feel free to implement some sort of `pair<string, string> splitAt(string, index)` if you feel so inclined.

_Note: Isn't it possible to create a `boost::asio`endpoint straight from a string?_

Reply:
Sorry, I don't see anything wrong with that line.

Reply:
What about now?
```cpp
const auto sepidx = address.find(':');                     // guaranteed not to throw
const std::string ipAddress  = address.substr(0, sepids);  // shorter
const std::string portNumber = address.substr(sepidx + 1); // no need to specify the end
```